v2/pkg/cli/delete.go

```diff
@@ -0,0 +1,309 @@
+	cmd.PersistentFlags().StringVarP(&opts.Global.ConfigPath, "config", "c", "", "Path to delete imageset configuration file")
+	cmd.PersistentFlags().StringVarP(&opts.Global.DeleteSource, "source", "s", "", "The working directory used to do the initial mirroring")
+	cmd.Flags().StringVar(&opts.Global.LogLevel, "loglevel", "info", "Log level one of (info, debug, trace, error)")
+	cmd.Flags().StringVar(&opts.Global.DeleteID, "delete-id", "", "Used to differentiate between versions for files created by the delete functionality")
+	cmd.Flags().StringVar(&opts.Global.DeleteYaml, "delete-yaml-file", "", "If set will use the generated or updated yaml file to delete contents")
+	cmd.Flags().BoolVar(&opts.Global.ForceCacheDelete, "force-cache-delete", ex.Opts.Global.ForceCacheDelete, "Used to force delete  the local cache manifests and blobs")
+	cmd.Flags().StringVar(&opts.Global.DeleteDestination, "destination", "", "The remote registry to delete from (optional)")
+	cmd.Flags().StringVar(&opts.Global.WorkingDir, "dir", workingDir, "Assets directory")
+	cmd.Flags().Uint16VarP(&opts.Global.Port, "port", "p", 55000, "HTTP port used by oc-mirror's local storage instance")
+	cmd.Flags().BoolVar(&opts.Global.V2, "v2", ex.Opts.Global.V2, "Redirect the flow to oc-mirror v2 - This is Tech Preview, it is still under development and it is not production ready.")
+	cmd.Flags().BoolVar(&opts.Global.DeleteGenerate, "generate", ex.Opts.Global.DeleteGenerate, "Used to generate the delete yaml for the list of manifests and blobs , used in the step to actually delete from local cahce and remote registry")
+	// nolint: errcheck
+	cmd.Flags().MarkHidden("v2")
+	cmd.Flags().AddFlagSet(&flagSharedOpts)
+	cmd.Flags().AddFlagSet(&flagRetryOpts)
+	cmd.Flags().AddFlagSet(&flagDepTLS)
+	cmd.Flags().AddFlagSet(&flagSrcOpts)
+	cmd.Flags().AddFlagSet(&flagDestOpts)
+
+	// hide flags
+	HideFlags(cmd)
+
+	return cmd
+}
+
+// Validate - cobra validation
+func (o ExecutorSchema) ValidateDelete() error {
+	if len(o.Opts.Global.DeleteSource) == 0 {
+		return fmt.Errorf("use the --source flag, it is mandatory when using the delete command")
+	} else {
+		if !strings.Contains(o.Opts.Global.DeleteSource, fileProtocol) {
+			return fmt.Errorf("--source flag must have a file:// protocol prefix")
+		}
+	}
+	if o.Opts.Global.DeleteGenerate {
+		if len(o.Opts.Global.ConfigPath) == 0 || o.Opts.Global.DeleteID == "" {
+			return fmt.Errorf("the --config and --delete-id flags are mandatory when used with the --generate flag")
+		}
+	} else {
+		if len(o.Opts.Global.DeleteYaml) == 0 || len(o.Opts.Global.DeleteDestination) == 0 {
+			return fmt.Errorf("the --delete-yaml-file and --destination flags are mandatory when not using the --generate flag")
+		}
+	}
+	if len(o.Opts.Global.DeleteDestination) > 1 && !strings.Contains(o.Opts.Global.DeleteDestination, dockerProtocol) {
+		return fmt.Errorf("--destination flag must have a docker:// protocol prefix")
+	}
+
+	// remove file protocol from delete source
```

Comment:
This doesn't really trim away the file:// protocol. In fact, validation fails if the input command arg contains `--delete-yaml-file file:///home/skhoury/clid20/working-dir/delete/delete-images-v4.13.yaml`

Reply:
Why - you dont do this for --config , i.e  --config file://isc.yaml , so why do you want to apply this rule here ?